osquery/database/tests/database_tests.cpp

```diff
@@ -130,7 +130,15 @@ static void ptree_to_json(const pt::ptree& pt_in, const JSON& rj_in) {
   EXPECT_TRUE(rj_out.fromString(conversion_out));
 
   /* Compare original rj with output */
-  EXPECT_EQ(rj_in.doc(), rj_out.doc());
+  std::string ptree_converted{""};
+  std::string rjin_expected{""};
+
+  auto s = rj_in.toString(rjin_expected);
+  EXPECT_TRUE(s.ok());
+  s = rj_out.toString(ptree_converted);
+  EXPECT_TRUE(s.ok());
+
+  EXPECT_EQ(ptree_converted, rjin_expected);
```

Comment:
Why compare the strings instead of the objects?

Reply:
It's my first impression that RJ doesn't have an `operator=` defined. We can add one if we like, but comparing these two objects directly wont give us the comparison we'd like, as it's attempting to see if the class object memory locations are the same, which they wont be. I think what we're wanting here is to see if the json data itself is the same, for which we'd need to access the string values. Doing the correct comparison as we had it will never pass as the class objects will have different memory locations, thus failing the test.